go/cmd/dolt/commands/log_graph.go

```diff
@@ -0,0 +1,428 @@
+// Copyright 2019 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package commands
+
+import (
+	"fmt"
+	"math"
+	"strings"
+
+	"github.com/dolthub/dolt/go/cmd/dolt/cli"
+	"github.com/dolthub/dolt/go/libraries/utils/argparser"
+	"github.com/dolthub/dolt/go/store/util/outputpager"
+)
+
+type CommitInfoWithChildren struct {
+	Commit           CommitInfo
+	Children         []string
+	X                int
+	Y                int
+	formattedMessage []string
+}
+
+var branchColors = []string{
+	"\033[31m", // Red
+	"\033[32m", // Green
+	"\033[34m", // Blue
+	"\033[35m", // Magenta
+	"\033[36m", // Cyan
+	"\033[39m", // Default
+}
+
+type BranchPathType struct {
+	Start int
+	End   int
+}
+
+// get the children of commits, and initialize the x and y coordinates of the commits
+func mapCommitsWithChildrenAndPosition(commits []CommitInfo) []*CommitInfoWithChildren {
+	childrenMap := make(map[string][]string)
+	for _, commit := range commits {
+		for _, parent := range commit.parentHashes {
+			childrenMap[parent] = append(childrenMap[parent], commit.commitHash)
+		}
+	}
+
+	var commitsWithChildren []*CommitInfoWithChildren
+	for y, commit := range commits {
+		commitsWithChildren = append(commitsWithChildren, &CommitInfoWithChildren{
+			Commit:   commit,
+			Children: childrenMap[commit.commitHash],
+			X:        -1,
+			// the y coordinate of the commit is initialized to the index of the commit as the commits are sorted
+			Y: y,
+		})
+	}
+
+	return commitsWithChildren
+}
+
+// wrap the commit message in a constrained width to better align the commit message with the graph
+func wrapTextOnWidth(text string, width int) []string {
```

Comment:
Awkward function name. `indentWithWrap` would be more straight forward

Reply:
Is wrapping word necessary? Does git do this? My suspicion is that it doesn't. The pager program will allow for right scrolling of the user wants it. I'm inclined to simplify the code. Indenting seems necessary, I just don't think that wrapping the words is required.